### PR TITLE
feature: allow scheduling of scheduled tasks

### PR DESCRIPTION
**Pull request by M-arcus**

### CODE REVIEW

**Comment by M-arcus** [comment]
image

**Comment by schneider-felix** [comment]
Doesn't really matter here but I wanted to point it out anyways. `Set task status to scheduled immediately` will not work when using the symfony scheduler bridge in shopware. The symfony scheduler stores its own `lastExecutionTime` and ignores the values from the database.

**Comment by shyim** [comment]
can we not overwrite the data of symfony scheduler?

**Comment by schneider-felix** [comment]
> can we not overwrite the data of symfony scheduler?

I don't think so. The only way to do this is to dispatch the message manually to the message bus. The scheduler keeps information about lastRun in a local variable. And with this PR https://github.com/shopware/shopware/pull/5456 it also writes it to cache. But it will never update the DB. 

On the other hand when using the symfony scheduler, the status will also never change to queued/failed. It will always remain in the scheduled state. This is because the bridge never updates this value. It just checks for it.

**Comment by M-arcus** [comment]
Maybe it should be dispatched immediately inside the `scheduleTask` function?
Like in https://github.com/shopware/shopware/blob/trunk/src/Core/Framework/MessageQueue/ScheduledTask/Scheduler/TaskScheduler.php#L147

**Comment by schneider-felix** [comment]
Nevermind, I was wrong about this altogether xD. It will always run, even if it is in the queued or failed state. The only exception is the inactive state. See here: https://github.com/shopware/shopware/blob/3d2e531639877ff0d839b98fc6081beec0cd5ebf/src/Core/Framework/MessageQueue/ScheduledTask/SymfonyBridge/ScheduleProvider.php#L46

**Comment by M-arcus** [comment]
@schneider-felix 

> Nevermind, I was wrong about this altogether xD. 

So the current state of the scheduling code in this PR is correct?

> It will always run, even if it is in the queued or failed state. 

Is this intended behaviour?

> The only exception is the inactive state

Sounds like new PR I can create :D `setInactive` or sth

File: src/Resources/app/administration/src/module/frosh-tools/component/frosh-tools-tab-scheduled/index.js [hunk 1]
```diff
@@ -104,6 +104,28 @@ Component.register('frosh-tools-tab-scheduled', {
...
+                    message: this.$tc('frosh-tools.scheduledTaskScheduleStarted', 0, {'name': item.name})
+                })
+                await this.froshToolsService.scheduleScheduledTask(item.id, immediately);
+                this.createNotificationSuccess({
+                    message: this.$tc('frosh-tools.scheduledTaskScheduleSucceed', 0, {'name': item.name})
+                })
+            } catch (e) {
+                this.createNotificationError({
+                    message: this.$tc('frosh-tools.scheduledTaskScheduleFailed', 0, {'name': item.name})
+                })
```
Comment (schneider-felix) [inline src/Resources/app/administration/src/module/frosh-tools/component/frosh-tools-tab-scheduled/index.js:121]:
```suggestion
                this.createNotificationInfo({
                    message: this.$t('frosh-tools.scheduledTaskScheduleStarted', {'name': item.name})
                })
                await this.froshToolsService.scheduleScheduledTask(item.id, immediately);
                this.createNotificationSuccess({
                    message: this.$t('frosh-tools.scheduledTaskScheduleSucceed', {'name': item.name})
                })
            } catch (e) {
                this.createNotificationError({
                    message: this.$t('frosh-tools.scheduledTaskScheduleFailed', {'name': item.name})
                })
```

You can use `$t` if you do not need pluralization. Then you can ommit the second parameter.

**Review by schneider-felix — CHANGES_REQUESTED** [review]

File: src/Resources/app/administration/src/module/frosh-tools/component/frosh-tools-tab-scheduled/index.js (hunk 1 above)
Comment (M-arcus) [reply src/Resources/app/administration/src/module/frosh-tools/component/frosh-tools-tab-scheduled/index.js:121]: This was copied from https://github.com/FriendsOfShopware/FroshTools/blob/main/src/Resources/app/administration/src/module/frosh-tools/component/frosh-tools-tab-scheduled/index.js#L85-L105

**Review by schneider-felix — APPROVED** [review]

**Pull request merged**